docs(tokens): add token documentation (#9232)
carbon-design-system/carbon: packages/styles/docs/sass.md

## Colors

> The colors package will give you access to every color token within the IBM Design Language.

Reviewer: ```suggestion
The colors package will give you access to each swatch from the IBM Design Language. You can refer to a color by its swatch and grade.
```

Access these color tokens for your project by using `@use`. For example: 

```scss

>  @use '@carbon/styles/scss/colors';

Reviewer: ```suggestion
 @use '@carbon/styles/scss/colors';
 
 .selector {
   background-color: colors.$blue-50;
 }
```

```

> | Import                               | Filepath            |

Reviewer: Make sure this is before all the content above

| :----------------------------------- | :------------------ |
| `@use '@carbon/styles/scss/colors';` | `scss/_colors.scss` |

> For more information checkout our [Files](#files) docs.

Reviewer: To see all the colors available to be imported, checkout our [colors](#todo) docs.

## Motion

The motion package provides helper functions, mixins, and duration tokens to add motion into your project.

> You can bring in the motion package and tokens by using `@use`. 
> 
> For example:
> 

Reviewer: I think we can drop these kinds of snippets and just reference the snippet directly 👀 

```suggestion
```

```scss
 @use '@carbon/styles/scss/motion';
 .dropdown--arrow {

>   transition: transform $duration-fast-02 motion(standard, productive);

Reviewer: I think the snippet will need to be updated to:

```scss
@use '@carbon/styles/scss/motion' as *;
```

For those variables to be defined.

 }
```
| Import                               | Filepath            |
| :----------------------------------- | :------------------ |
| `@use '@carbon/styles/scss/motion';` | `scss/_motion.scss` |

> For more information checkout our [Files](#files) docs.

Reviewer: ```suggestion
For more information, checkout our [motion](#todo) docs.
```

## Type

The type package entrypoint allows you to specifically bring type tokens into your project. The type package includes various type tokens and mixins.

> To bring in the `@carbon/type` package use `@use`. For example: 

Reviewer: ```suggestion
```

```scss

>  @use '@carbon/styles/scss/type';

Reviewer: ```suggestion
 @use '@carbon/styles/scss/type';
 
 .selector {
   @include type.style(type.$productive-heading-01);
 }
```

```

| Import                             | Filepath          |
| :--------------------------------- | :---------------- |
| `@use '@carbon/styles/scss/type';` | `scss/_type.scss` |

> For more information checkout our [Files](#files) docs.

Reviewer: ```suggestion
For more information, check out our [type](#todo) docs.
```
